### PR TITLE
initial setup for frontend

### PR DESCRIPTION
**Pull request by Alexander-Wen**
still need to set up the redux part of it, but for now this is working and shows a black screen. If you see a black screen when you go to localhost:3000, then that means everything is working :smile:

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const proxy = httpProxy.createProxyServer(proxyOptions);
+
+app.use('/api', (req, res, next) => {
+  req.url = '/api' + req.url;
+  return proxy.proxyRequest(req, res);
+});
+
+app.use(serveWebpackClient({
+  distPath: path.join(__dirname, 'public'),
```
Comment (Alexander-Wen) [inline index.js:22]: compiled code goes into /public

File: package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "reflect",
+  "version": "1.0.0",
+  "description": "Reflect is a smart mirror made to streamline your morning",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "NODE_ENV=production webpack -p",
```
Comment (Alexander-Wen) [inline package.json:8]: NODE_ENV is default dev

File: src/containers/HomeContainer.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+class HomeContainer extends React.Component {
+  render() {
+    return (
+      <div>
+        dank memes
```
Comment (Alexander-Wen) [inline src/containers/HomeContainer.js:7]: this wont show yet, because ive yet to do some stuff, hence the TODOs

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  }),
+];
+
+const plugins = basePlugins
+  .concat(process.env.NODE_ENV === 'production' ? prodPlugins : [])
+  .concat(process.env.NODE_ENV === 'development' ? devPlugins : []);
+
+module.exports = {
+  entry: {
+    app: getEntrySources(['./src/app.js']),
```
Comment (Alexander-Wen) [inline webpack.config.js:45]: this is how the app knows where to enter from

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      },
+    ],
+    loaders: [
+      {
+        test: /\.css$/,
+        loader: 'style-loader!css-loader!postcss-loader!cssnext-loader',
+      },
+      {
+        test: /\.js$/,
+        loaders: ['react-hot', 'babel'],
```
Comment (Alexander-Wen) [inline webpack.config.js:74]: needed for babel

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  .concat(process.env.NODE_ENV === 'production' ? prodPlugins : [])
+  .concat(process.env.NODE_ENV === 'development' ? devPlugins : []);
+
+module.exports = {
+  entry: {
+    app: getEntrySources(['./src/app.js']),
+  },
+
+  output: {
+    path: path.join(__dirname, 'public'),
```
Comment (Alexander-Wen) [inline webpack.config.js:49]: goes to public folder

File: src/store/config/routes.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Route } from 'react-router';
+import HomeContainer from '../../containers/HomeContainer';
+
+const createRoutes = (store) => {
+  return (
+    <Route path="/" component={HomeContainer}/>
```
Comment (Alexander-Wen) [inline src/store/config/routes.js:7]: all future routes will go here as well

File: src/app.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import createStore from './store/configureStore';
```
Comment (Alexander-Wen) [inline src/app.js:4]: configureStore does nothing rn, so this will give an error, ignore it for now

File: package.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "reflect",
+  "version": "1.0.0",
+  "description": "Reflect is a smart mirror made to streamline your morning",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (Alexander-Wen) [inline package.json:7]: will put tests in later...maybe

File: index.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
+const express = require('express');
+const httpProxy = require('http-proxy');
+const path = require('path');
+const serveWebpackClient = require('serve-webpack-client');
+const app = express();
+
+const PORT = process.env.PORT || 3000;
+const API_URL = process.env.API_URL || 'https://project-reflect-api.herokuapp.com/';
```
Comment (Alexander-Wen) [inline index.js:8]: will connect to live backend once its live

**Comment by AdrianMachado** [comment]
shipit

**Pull request merged**